go-controller/pkg/ovn/pods.go

```diff
@@ -140,11 +141,16 @@ func (oc *DefaultNetworkController) addLogicalPort(pod *kapi.Pod) (err error) {
 		return nil
 	}
 
-	_, network, err := util.PodWantsMultiNetwork(pod, oc.NetInfo)
+	_, networkMap, err := util.PodWantsNetwork(pod, oc.NetInfo)
 	if err != nil {
 		// multus won't add this Pod if this fails, should never happen
 		return fmt.Errorf("error getting default-network's network-attachment for pod %s/%s: %v", pod.Namespace, pod.Name, err)
 	}
+	// for default network, networkMap either is empty or contains only one network selection element
+	var network *nadapi.NetworkSelectionElement
+	for _, network = range networkMap {
+		break
+	}
```

Comment:
nit: would it be better to rather check this is empty or has a single item and return an error otherwise.

Additionally, are we validating somewhere that the default-network annotation actually points to a NAD attaching to the primary network. I guess this should fail somewhere with a meaningful error.

Reply:
then it is probably best for nad controller to monitor default nad as well, and let the controller also has nadNames sync.Map so it can be validated by HasNAD() function. That is how it is done in downstream as well.

If you think that is the way to go, I'd rather leave it as a later PR. Before we don't valid NAD for default network either, this PR does not make it worse.

Reply:
for this first part fo the comments, the way how default NAD is defined determined that the networkMap is either empty or has single NAD (either "v1.multus-cni.io/default-network" is defined or not). Do we still need to check it here?

Reply:
> then it is probably best for nad controller to monitor default nad as well, so the default network controller also has nadNames sync.Map so it can be validated by HasNAD() function. That is how it is done in downstream as well.
> 
> If you think that is the way to go, I'd rather leave it as a later PR. Before we don't valid NAD for default network either, this PR does not make it worse.

I think it could be simpler than that, we could just check that the NAD is of primary type (whatever criteria we are using to determine that) in `GetK8sPodDefaultNetworkSelection`. We can leave it for a later PR if you think it does not fit this PR.

Reply:
> for this first part fo your comment, the way how default NAD is defined determined that the networkMap is either empty or has single NAD (either "v1.multus-cni.io/default-network" is defined or not). Do we still need to check it here?

I just meant adding something like
```
if len(networkMap) > 1 {
  return fmt.Errorf(....)
}
```
but I guess we can assume we will only ever get one network for the default as there is no possible way API wise to specify multiple different NADs for the default network.

Reply:
i think for the sake of this code we should have an error check like the above. We shouldn't depend on API or code from other places to catch an error in this function. It's better to have more robust error checking where possible.